lib/destroy-session.js

```diff
@@ -0,0 +1,18 @@
+'use strict'
+
+module.exports = function destroySession (done) {
```

Comment:
an alternative approach to this PR is to do what this function does (put the store's methods directly on `request` and encapsulate the session ID), and make `sessionStore` remain what the user passed in (i.e. callback only).

e.g. `request.sessionStore.set(request.session.sessionId, {})` could just be `request.setSession({})`, `request.sessionStore.get(request.session.sessionId)` could be `request.getSession()`.

Reply:
or even stick more methods on the `session` object itself rather than directly on the `request` object - today it has only `regenerate` and `touch`, but not `save` etc.. When using Express' `session` I never interacted with `sessionStore` directly, all methods (`save`, `regenerate`, `reload`, `destroy`) were all on the session object.

Reply:
@SimenB Thank you for the input! My only gripe with putting methods on the session is that it "occupies" the names that you use on `request.session`. While it is technically possible to do `request.session.destroy = someValue` (since the methods live on the session class's prototype object), it should be avoided as it causes confusion.

Side note:  
I like the approach that [fastify-secure-session](https://github.com/fastify/fastify-secure-session) went with, that to set session values you must use `request.session.set('name', 'value')`. Which enables keeping track if the session is changed with a simple [flag](https://github.com/fastify/fastify-secure-session/blob/ee17469d15472614b47f9390c806cd9329b4a6f1/index.js#L204) and avoid [this](https://github.com/fastify/session/blob/4546ff96cd3de71a4f33e6b11522adc5f8a3ad6b/lib/fastifySession.js#L211). But this means a breaking change on the session interface.

Reply:
`touch` and `regenerate` is already on `session`, as my table shows the current API is _very_ inconsistent.

(I personally have nothing against going through a setter rather than using assignment, hugely breaking for this module, tho)

Reply:
That is indeed a huge breaking change, and I'm not sure if end users of this library would appreciate. I think it's in everyone's best interest to fix the inconsistency without breaking too many things, or at least having a clear way on migrating if we decide that the setter syntax is the way to go.

If this library uses the setter syntax that would also solve [this issue](https://github.com/fastify/fastify-passport/issues/319) from fastify-passport. I had been trying to make fastify-passport work with a storage-backed session middleware (with the goal of migrating projects from express to fastify) so I started building [this](https://github.com/lokshunhung/fastify-compat-session) but did not have the time to finish it. So I guess it's better to pool efforts collectively from the community than to roll things on my own.

Reply:
@mcollina What are your opinions on this?

Reply:
I think this topic can open a new issue for it. I support to unify the current API.
And it is not really related to this PR.

Reply:
Right, but by unifying this PR would probably be completely different

Reply:
I created a new issue for this: #34 

Can I mark this as resolved?